src/api-service/__app__/onefuzzlib/agent_authorization.py

```diff
@@ -4,77 +4,50 @@
 # Licensed under the MIT License.
 
 import logging
-from typing import Callable, Union
+from typing import Callable
 from uuid import UUID
 
 import azure.functions as func
-import jwt
 from memoization import cached
 from onefuzztypes.enums import ErrorCode
-from onefuzztypes.models import Error
-from pydantic import BaseModel
+from onefuzztypes.models import Error, UserInfo
 
 from .azure.creds import get_scaleset_principal_id
-from .pools import Scaleset
+from .pools import Pool, Scaleset
 from .request import not_ok
+from .user_credentials import parse_jwt_token
 
 
-class TokenData(BaseModel):
-    application_id: UUID
-    object_id: UUID
-
-
-def try_get_token_auth_header(request: func.HttpRequest) -> Union[Error, TokenData]:
-    """ Obtains the Access Token from the Authorization Header """
-    auth: str = request.headers.get("Authorization", None)
-    if not auth:
-        return Error(
-            code=ErrorCode.INVALID_REQUEST, errors=["Authorization header is expected"]
-        )
-    parts = auth.split()
-
-    if parts[0].lower() != "bearer":
-        return Error(
-            code=ErrorCode.INVALID_REQUEST,
-            errors=["Authorization header must start with Bearer"],
-        )
-
-    elif len(parts) == 1:
-        return Error(code=ErrorCode.INVALID_REQUEST, errors=["Token not found"])
-
-    elif len(parts) > 2:
-        return Error(
-            code=ErrorCode.INVALID_REQUEST,
-            errors=["Authorization header must be Bearer token"],
-        )
+@cached(ttl=60)
+def is_agent(token_data: UserInfo) -> bool:
 
-    # This token has already been verified by the azure authentication layer
-    token = jwt.decode(parts[1], verify=False)
-    return TokenData(application_id=UUID(token["appid"]), object_id=UUID(token["oid"]))
+    if token_data.object_id:
+        # backward compatibility case for scalesets deployed before the migration
```

Comment:
Can you add more detail to this comment, emphasizing when we can remove this?

We should probably have something greppable for this back-compat checks.

Reply:
I don't actually know what's required here.  This was just lifted entirely from the agent authorization code.

While that's reasonable to do for the future, out of scope for this PR.